front/lib/models/assistant/agent_step_content.ts

```diff
@@ -0,0 +1,104 @@
+import type { CreationOptional, ForeignKey, NonAttribute } from "sequelize";
+import { DataTypes } from "sequelize";
+
+import { AgentMessage } from "@app/lib/models/assistant/conversation";
+import { frontSequelize } from "@app/lib/resources/storage";
+import { WorkspaceAwareModel } from "@app/lib/resources/storage/wrappers/workspace_models";
+import type { AssistantContentItemType } from "@app/types/assistant/agent_message_content";
+
+export class AgentStepContentModel extends WorkspaceAwareModel<AgentStepContentModel> {
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare agentMessageId: ForeignKey<AgentMessage["id"]>;
+  declare step: number;
+  declare indexInStep: number;
+  declare type: AssistantContentItemType["type"];
+  declare value: AssistantContentItemType;
+
+  declare agentMessage?: NonAttribute<AgentMessage>;
+}
+
+AgentStepContentModel.init(
+  {
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    agentMessageId: {
+      type: DataTypes.BIGINT,
+      allowNull: false,
+      references: {
+        model: "agent_messages",
+        key: "id",
+      },
+    },
+    step: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    indexInStep: {
```

Comment:
hurts my eyes. Can we get away with `index` :grimacing: ?

Reply:
sure !